### PR TITLE
{data}[foss/2024a] GDAL v3.10.0

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

* [x] #21608

### CODE REVIEW

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/9f2a15f4068fc791fb077074882344ee for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0210u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8480CL (sapphirerapids), Python 3.6.8
See https://gist.github.com/branfosj/176c4c87a72779f811ccfab9b0e65de2 for a full test report.

**Comment by smoors** [comment]
>easybuild.tools.build_log.EasyBuildError: 'Missing dependencies: Armadillo/14.0.3-foss-2024a (no easyconfig file or existing module found)'

**Comment by branfosj** [comment]
> > easybuild.tools.build_log.EasyBuildError: 'Missing dependencies: Armadillo/14.0.3-foss-2024a (no easyconfig file or existing module found)'

Is in #21608 but that also has a different GDAL.

File: easybuild/easyconfigs/g/GDAL/GDAL-3.10.0-foss-2024a.eb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    ('PCRE', '8.45'),
+    ('PROJ', '9.4.1'),
+    ('libgeotiff', '1.7.3'),
+    ('SciPy-bundle', '2024.05'),
+    ('HDF5', '1.14.5'),
+    ('HDF', '4.3.0'),
+    ('Armadillo', '14.0.3'),
+    ('CFITSIO', '4.4.1'),
+    ('zstd', '1.5.6'),
+    ('giflib', '5.2.2'),
```
Comment (smoors) [inline easybuild/easyconfigs/g/GDAL/GDAL-3.10.0-foss-2024a.eb:50]: can we use v5.2.1 here instead? is already used as dep for
>{'ASE-3.23.0-gfbf-2024a.eb', 'matplotlib-3.9.2-gfbf-2024a.eb', 'modin-0.32.0-foss-2024a.eb', 'GPAW-24.6.0-foss-2024a.eb', 'Seaborn-0.13.2-gfbf-2024a.eb', 'dask-2024.9.1-gfbf-2024a.eb', 'bokeh-3.6.0-gfbf-2024a.eb', 'nf-core-2.14.1-foss-2024a.eb'}

Comment (branfosj) [reply easybuild/easyconfigs/g/GDAL/GDAL-3.10.0-foss-2024a.eb]: 3ed0dcf

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen3

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/cbb59296664c44688598fa173932154c for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
bear-pg0210u03b.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8480CL (sapphirerapids), Python 3.6.8
See https://gist.github.com/branfosj/3011e60158b93dd1f514540c2a9eebdb for a full test report.

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @branfosj!

**Pull request merged**